File: src/面试题17_10.find-majority-element-lcci.1000038/solution.ts
function majorityElement (nums: number[]): number {
    if (nums.length === 0) {
        return -1;
    }

    let result = nums[0];
    let count = 1;
    for (let i = 1; i < nums.length; i++) {
        if (nums[i] === result) {
            count++;
        } else if (count > 0) {
            count--;
        } else {
            count = 1;
            result = nums[i];
        }
    }
    count = 0;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] === result) {
            count++;
        }
    }
    if (count > nums.length / 2) {
        return result;
    }
    return -1;
}

File: src/面试题17_10.find-majority-element-lcci.1000038/solution.java
class Solution {
    public int majorityElement(int[] nums) {
        int count = 1;
        int element = nums[0];
        for(int i=1;i<nums.length;i++){
            if(nums[i] == element){
                count++;
            }else if(count>0){
                count--;
            }else{
                count = 1;
                element = nums[i];
            }
        }

        count = 0;
        for(int n:nums){
            if(n == element){
                count++;
            }
        }
        if(count>nums.length/2){
            return element;
        }

        return -1;
    }
}